plugins/outputs/kairosdb/kairosdb.go

```diff
@@ -0,0 +1,287 @@
+			}
+
+			metric := &HttpMetric{
+				Metric: sanitize(fmt.Sprintf("%s%s%s%s",
+					o.Prefix, m.Name(), o.Separator, fieldName)),
+				Tags:      tags,
+				Timestamp: now,
+				Value:     value,
+			}
+
+			if err := http.sendDataPoint(metric); err != nil {
+				return err
+			}
+		}
+	}
+
+	if err := http.flush(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *KairosDB) WriteTelnet(metrics []telegraf.Metric, u *url.URL) error {
+	// Send Data with telnet / socket communication
+	uri := fmt.Sprintf("%s:%d", u.Host, o.Port)
+	tcpAddr, _ := net.ResolveTCPAddr("tcp", uri)
+	connection, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		return fmt.Errorf("KairosDB: Telnet connect fail")
+	}
+	defer connection.Close()
+
+	for _, m := range metrics {
+		now := m.Time().UnixNano() / 1000000
+		tags := ToLineFormat(cleanTags(m.Tags()))
+
+		for fieldName, value := range m.Fields() {
+			switch value.(type) {
+			case int64:
+			case uint64:
+			case float64:
+			default:
+				log.Printf("D! KairosDB does not support metric value: [%s] of type [%T].\n", value, value)
+				continue
+			}
+
+			metricValue, buildError := buildValue(value)
+			if buildError != nil {
+				log.Printf("E! KairosDB: %s\n", buildError.Error())
+				continue
+			}
+
+			messageLine := fmt.Sprintf("put %s %v %s %s\n",
+				sanitize(fmt.Sprintf("%s%s%s%s", o.Prefix, m.Name(), o.Separator, fieldName)),
+				now, metricValue, tags)
+
+			_, err := connection.Write([]byte(messageLine))
+			if err != nil {
+				return fmt.Errorf("KairosDB: Telnet writing error %s", err.Error())
+			}
+		}
+	}
+
+	return nil
+}
+
+func cleanTags(tags map[string]string) map[string]string {
+	tagSet := make(map[string]string, len(tags))
+	for k, v := range tags {
+		val := sanitize(v)
+		if val != "" {
+			tagSet[sanitize(k)] = val
+		}
+	}
+	return tagSet
+}
+
+func buildValue(v interface{}) (string, error) {
+	var retv string
+	switch p := v.(type) {
+	case int64:
+		retv = IntToString(int64(p))
+	case uint64:
+		retv = UIntToString(uint64(p))
+	case float64:
+		retv = FloatToString(float64(p))
+	default:
+		return retv, fmt.Errorf("unexpected type %T with value %v for KairosDB", v, v)
+	}
+	return retv, nil
+}
+
+func IntToString(input_num int64) string {
+	return strconv.FormatInt(input_num, 10)
+}
+
+func UIntToString(input_num uint64) string {
+	return strconv.FormatUint(input_num, 10)
+}
+
+func FloatToString(input_num float64) string {
+	return strconv.FormatFloat(input_num, 'f', 6, 64)
+}
+
+func (o *KairosDB) SampleConfig() string {
+	return sampleConfig
+}
+
+func (o *KairosDB) Description() string {
+	return "Configuration for KairosDB server to send metrics to"
+}
+
+func (o *KairosDB) Close() error {
+	return nil
+}
+
+func sanitize(value string) string {
+	// Apply special hypenation rules to preserve backwards compatibility
+	value = hypenChars.Replace(value)
+	// Replace any remaining illegal chars
+	return allowedChars.ReplaceAllLiteralString(value, "_")
+}
```

Comment:
Can you please move this function above `SampleConfig()` to have the plugin specific ones grouped.